src/analysis/processing/qgsalgorithmpromotetomultipart.cpp

```diff
@@ -65,6 +66,12 @@ QgsPromoteToMultipartAlgorithm *QgsPromoteToMultipartAlgorithm::createInstance()
   return new QgsPromoteToMultipartAlgorithm();
 }
 
+bool QgsPromoteToMultipartAlgorithm::supportInPlaceEdit( const QgsVectorLayer *layer ) const
+{
+  Q_UNUSED( layer );
+  return false;
```

Comment:
i wonder about this one -- maybe we should allow it, if layer is multipart

My rationale here: sometimes i do some edit operation, but i'm blocked from saving the changes to a provider because the edit op has resulted in a conversion from multi -> single part. If we allowed this algorithm, you could promote to multipart in place and then you'd be able to save!